src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructor.cs

```diff
@@ -178,5 +184,27 @@ protected override bool ShouldBindAttributes(AttributeListSyntax attributeDeclar
 
             return false;
         }
+
+        public Roslyn.Utilities.IReadOnlySet<ParameterSymbol> GetParametersPassedToTheBase()
+        {
+            if (_parametersPassedToTheBase != null)
+            {
+                return _parametersPassedToTheBase;
+            }
+
+            TryGetBodyBinder().BindConstructorInitializer(GetSyntax().PrimaryConstructorBaseTypeIfClass, BindingDiagnosticBag.Discarded);
+
+            if (_parametersPassedToTheBase is null)
+            {
+                _parametersPassedToTheBase = SpecializedCollections.EmptyReadOnlySet<ParameterSymbol>();
+            }
+
+            return _parametersPassedToTheBase;
+        }
+
+        internal void SetParametersPassedToTheBase(Roslyn.Utilities.IReadOnlySet<ParameterSymbol> value)
+        {
+            _parametersPassedToTheBase = value;
```

Comment:
>_parametersPassedToTheBase [](http://example.com/codeflow?start=12&length=26)

Consider adding `Debug.Assert(_parametersPassedToTheBase is null);`

Reply:
> Consider adding Debug.Assert(_parametersPassedToTheBase is null);

That would be a wrong thing to do. It is possible that another thread already set the value. I do not use InterlockExchange because we don't care about identity of the set itself